tests/test_pipelines_text_classification.py

```diff
@@ -24,3 +29,104 @@ class TextClassificationPipelineTests(MonoInputPipelineCommonMixin, unittest.Tes
+                _classifier = pipeline(task="sentiment-analysis", model=model, tokenizer=tokenizer)
+                _string_output = _classifier(string_input)
+                _string_list_output = _classifier(string_list_input)
+                check_output(_string_output)
+                return _string_output, _string_list_output
+
+            string_output = classifier(string_input)
+            string_list_output = classifier(string_list_input)
+
+            string_output_default = pipeline_call_argument("default")
+            string_output_sigmoid = pipeline_call_argument("sigmoid")
+            string_output_softmax = pipeline_call_argument("softmax")
+            string_output_none = pipeline_call_argument("none")
+
+            string_output_init_default = pipeline_init_argument("default")
+            string_output_init_sigmoid = pipeline_init_argument("sigmoid")
+            string_output_init_softmax = pipeline_init_argument("softmax")
+            string_output_init_none = pipeline_init_argument("none")
+
+            string_output_model_default = pipeline_model_argument("default")
+            string_output_model_sigmoid = pipeline_model_argument("sigmoid")
+            string_output_model_softmax = pipeline_model_argument("softmax")
+            string_output_model_none = pipeline_model_argument("none")
+
+            should_be_equal = [
+                (
+                    (string_output, string_list_output),
+                    string_output_default,
+                    string_output_init_default,
+                    string_output_model_default,
+                ),
+                (string_output_sigmoid, string_output_init_sigmoid, string_output_model_sigmoid),
+                (string_output_softmax, string_output_init_softmax, string_output_model_softmax),
+                (string_output_none, string_output_init_none, string_output_model_none),
+            ]
+
+            for tuples_containing_equal_values in should_be_equal:
+                # Retrieve each tuple from the list
+                for tuple_value_0 in tuples_containing_equal_values:
+                    for tuple_value_1 in tuples_containing_equal_values:
+                        if tuple_value_0 is not tuple_value_1:
+                            # Compare each tuple value with all the others, as long as they're not the same object
+                            for pipeline_output_0, pipeline_output_1 in zip(tuple_value_0, tuple_value_1):
+                                # Compare all outputs (call, init, model argument)
+                                for example_result_0, example_result_1 in zip(pipeline_output_0, pipeline_output_1):
+                                    # Iterate through the results
+                                    self.assertTrue(
+                                        np.allclose(example_result_0["score"], example_result_1["score"], atol=1e-6)
+                                    )
```

Comment:
I think those tests are super hard to read and to maintain. Because it contains  lot of logic, determining of this test is correct or not is kind of hard.
Also the expected values are not hardcoded in the test meaning  drift in output vlues would not be caught by this test.

I am a proponent of much simpler tests in logic, but much more verbose.

```python
classifier = pipeline(task="sentiment-analysis", model=model, tokenizer=tokenizer)

self.assertEqual(nested_simplify(classifier("I really disagree with what you've said.")), [{'score': 0.1, 'label' : "LABEL_0"}])

# other function to apply
self.assertEqual(nested_simplify(classifier("I really disagree with what you've said.", function_to_apply="sigmod")), [{'score': 0.2, 'label' : "LABEL_0"}])

# function_to_apply as pipeline arg
classifier2 = pipeline(task="sentiment-analysis", model=model, tokenizer=tokenizer, function_to_apply="sigmoid")
self.assertEqual(nested_simplify(classifier("I really disagree with what you've said.")), [{'score': 0.2, 'label' : "LABEL_0"}])

# from config
# check default strategy depends on  num_labels
# ....
```

I would happy to make a separate PR to suggest this and be compliant with the current written tests.

@LysandreJik Do you agree ?